### PR TITLE
[DOCS] Add "version" to template and pipeline docs

### PR DESCRIPTION
**Pull request by pickypg**
This adds details about the `"version"` field to both the template and pipeline pages.

### CODE REVIEW

File: docs/reference/ingest/ingest-node.asciidoc [hunk 1]
```diff
@@ -90,6 +90,56 @@ For each returned pipeline, the source and the version are returned.
...
 You can specify multiple IDs to return more than one pipeline. Wildcards are also supported.
 
+[float]
+[[versioning-pipelines]]
+==== Pipeline Versioning
+
+Pipelines can optionally add a `version` number, which can be any integer value,
+in order to simplify pipeline management by external systems. The `version`
+field is completely optional and it is meant solely for external management of
+pipelines. To unset a `version`, simply replace the template without specifying
```
Comment (tlrx) [inline docs/reference/ingest/ingest-node.asciidoc]: template -> pipeline

**Review by tlrx — APPROVED** [review]
LGTM, left one minor comment

**Comment by clintongormley** [comment]
@pickypg what happened here?

**Comment by pickypg** [comment]
@clintongormley I was letting it run through the tests in CI, then naturally forgot about it. :)

I noticed that you removed the 5.0 tag, but since it's docs, I can still backport this, right?

**Comment by pickypg** [comment]
5.x: 4f7d69a704e6b707bc233b65125159590cd4ef5a
5.0: 0ba1f251bce1026c64ecebf2dc750e6616c98735

**Pull request merged**